File: Selection_Sort/SelectionSort.java
import java.util.Arrays;

public class SelectionSort {
    
    public static void main(String[] args) {
        int[] arr = {8, 10, 7, 1, 4, 6, 2, 5, 9, 3};
        selectionSort(arr);

        System.out.println(Arrays.toString(arr));
    }

    public static void selectionSort(int[] arr) {
        int n = arr.length;

        for (int i = 0; i < n; i++) {
            int min = i;

            for (int j = min + 1; j < n; j++) {
                if (arr[j] < arr[min]) min = j;
            }

            int temp = arr[i];
            arr[i] = arr[min];
            arr[min] = temp;
        }
    }
}

File: Selection_Sort/selection_sort.py
def selection_sort(arr: list):
    n = len(arr)

    for i in range(0, n):
        min = i

        for j in range(min + 1, n):
            if arr[j] < arr[min]:
                min = j

        arr[i], arr[min] = arr[min], arr[i]


arr = [8, 10, 7, 1, 4, 6, 2, 5, 9, 3]
selection_sort(arr)

print(arr)
